### ISSUE TITLE
Wrong source number used in answer

### ISSUE THREAD

**Issue by ZainGithub12** [miurla/morphic#280]
@miurla A source number used in the answer was not right. It says 2, but it should say 3. 

You can see in source list that the 3rd source is about 6th anniversary

![Screenshot_20240728_195630_Chrome.jpg](https://github.com/user-attachments/assets/143bb8a4-dea7-44d1-b8ce-3c3bc36e84ff)

This is talking about 6th anniversary but says source 2 next to the answer

![Screenshot_20240728_195635_Chrome.jpg](https://github.com/user-attachments/assets/8ee3cbba-d5cc-4f1c-9ade-2e3ee9fb72f0)

**Comment by miurla** [comment]
Thank you for reporting this. We will check and improve this mode.

**Comment by ZainGithub12** [comment]
@miurla, When it's resolved, please let me know. 👍

**Comment by miurla** [comment]
@ZainGithub12 

> (Quality) GPT mode used.

Could you please check again to see if this mode can be reproduced now that it has been reverted to the previous version?

**Comment by ZainGithub12** [comment]
@miurla, Yes, it gives the right source number for the sixth anniversary now.

![Screenshot_20240808_082147_Chrome.jpg](https://github.com/user-attachments/assets/85abe509-ac13-4d91-b1f2-1002975c32c7)
